server/api/controllers/target-state/targetStateController.js

```diff
@@ -2,8 +2,22 @@
 
 'use strict';
 
+let co = require('co');
 let GetServerRoles = require('queryHandlers/services/GetServerRoles');
 
+function scanAndDelete(environment, parameters) {
+  return co(function* () {
+    let keyValuePairs = yield serviceTargets.getTargetState(query.environment, { key: parameters.keyPrefix, recurse: true });
+    let erasedKeys = yield keyValuePairs.filter((keyValuePair) =>
+      parameters.condition(keyValuePair.key, keyValuePair.value)
+    ).map((keyValuePair) => {
+      return serviceTargets.removeTargetState(environment, { key: keyValuePair.key }).then(() => keyValuePair.key);
+    });
+
+    return erasedKeys;
+  });
+}
+
```

Comment:
Can we keep business / implementation logic somewhere separate and reusable (eg in a command) - otherwise logic for deleting keys is coupled to our API?